### PR TITLE
Config proto should be passed by reference.

### PR DESCRIPTION
**Pull request by michelle192837**

### CODE REVIEW

**Comment by chases2** [comment]
The change seems fine, but what's the reasoning for this change? In Go, moving a constant struct from value to reference [isn't universally faster](https://medium.com/@vCabbage/go-are-pointers-a-performance-optimization-a95840d3ef85).

/lgtm
/hold

File: config/config.go [hunk 1]
```diff
@@ -155,15 +155,15 @@ func validateAllUnique(c configpb.Configuration) error {
 	return mErr
 }
 
-func validateReferencesExist(c configpb.Configuration) error {
+func validateReferencesExist(c *configpb.Configuration) error {
```
Comment (chases2) [inline config/config.go]: If these function arguments are going to change to pointers, they now need null checks.

```
if c == null {
  return fmt.Errorf("descriptive message")
}
```

Similarly for other functions.

**Comment by michelle192837** [comment]
The change is because of apparent potential issues with shallow-copying proto messages, not performance.

File: config/config.go (hunk 1 above)
Comment (michelle192837) [reply config/config.go]: Fiiiiiiixed I think?

Comment (michelle192837) [reply config/config.go]: In this PR: Michelle learned the importance of using the dang proto methods and not just accessing things willy-nilly.

**Pull request merged**